src/core/CausalTree.ts

```diff
@@ -0,0 +1,58 @@
+import Atom from './Atom';
+import AtomId from './AtomId';
+import IndexMap from './IndexMap';
+import { getNewUuid } from '../utils/functions';
+
+/**
+ * Returns the index where a site is (or should be) in the sitemap.
+ * Time complexity of O(log(sitemap.length)).
+ * @param sitemap
+ * @param siteUuid
+ */
+const findSiteIndex = (sitemap: string[], siteUuid: string): number => {
+  let left = 0;
+  let right = sitemap.length - 1;
+  while (left <= right) {
+    const mid = Math.floor((left + right) / 2);
+    if (sitemap[mid] === siteUuid) return mid;
+    if (sitemap[mid] < siteUuid) left = mid + 1;
```

Comment:
Do we have a linter rule preventing these "naked" if-else from ever be used outside a one-liner?

From my C days I was trained to always use "braced" if-else. A famous example happened on an Apple product that had the following code:

```c
if (!ptr)
  goto fail;
  goto fail;
```
The second line was copied accidentally, and then a critical function would always jump to the 'fail' label instead of running its logic.

Reply:
Yes, [this linter rule](https://eslint.org/docs/latest/rules/curly) checks for braces on control statements.